src/main/java/org/opensearch/knn/index/VectorDataType.java

```diff
@@ -31,6 +35,18 @@ public enum VectorDataType {
         public FieldType createKnnVectorFieldType(int dimension, VectorSimilarityFunction vectorSimilarityFunction) {
             return KnnByteVectorField.createFieldType(dimension, vectorSimilarityFunction);
         }
+
+        @Override
+        public float[] getValue(BytesRef binaryValue) {
+            float[] vector = new float[binaryValue.length];
```

Comment:
Why use a float[] for byte type? Shouldnt it be an int[]?

Reply:
We are using this method for scripting functions to retrieve the vector from docValues to calculate the score. SO, it doesn't make any difference if we return it as int[] or float[]. If we return it as int[] then again we need to do some method overloading and add methods for the spacetype functions in ScoringUtils to accept int[].